validator/src/consensus/storage/sqlite.ts

```diff
@@ -47,17 +63,29 @@ export class SqliteStorage
 			CREATE TABLE IF NOT EXISTS groups(
 				id TEXT NOT NULL,
 				threshold INTEGER NOT NULL,
-				public_key TEXT,
+				public_key BLOB,
 				PRIMARY KEY(id)
 			);
 
 			CREATE TABLE IF NOT EXISTS group_participants(
 				group_id TEXT NOT NULL,
 				id INTEGER NOT NULL,
 				address TEXT NOT NULL,
-				verification_share TEXT,
-				signing_share TEXT,
+				coefficients BLOB,
+				commitments BLOB,
+				verification_share BLOB,
+				signing_share BLOB,
 				PRIMARY KEY(group_id, id),
+				FOREIGN KEY(group_id) REFERENCES groups(id) ON DELETE CASCADE,
+				UNIQUE(group_id,address)
+			);
+
+			CREATE TABLE IF NOT EXISTS group_secret_shares(
+				group_id TEXT NOT NULL,
+				address TEXT NOT NULL,
+				from_participant INTEGER NOT NULL,
+				secret_share BLOB NOT NULL,
```

Comment:
did you have a specific reason why you used blobs for secret_share (and some other bytes based data), but not for group_id and address (so some other bytes based data)?

Reply:
Makes sense